library/alloc/src/string.rs

```diff
@@ -1849,6 +1849,34 @@ impl String {
         let slice = self.vec.into_boxed_slice();
         unsafe { from_boxed_utf8_unchecked(slice) }
     }
+
+    /// Consumes and leaks the `String`, returning a mutable reference to the contents,
+    /// `&'a mut str`.
+    ///
+    /// This function is mainly useful for data that lives for the remainder of
+    /// the program's life. Dropping the returned reference will cause a memory
+    /// leak.
+    ///
+    /// # Examples
+    ///
+    /// Simple usage:
+    ///
+    /// ```
+    /// #![feature(string_leak)]
+    ///
+    /// pub fn main() {
+    ///     let x = String::from("bucket");
+    ///     let static_ref: &'static mut str = x.leak();
+    ///     assert_eq!(static_ref, "bucket");
+    /// }
+    /// ```
+    #[cfg(not(no_global_oom_handling))]
```

Comment:
```suggestion
    #[cfg(not(no_global_oom_handling))]
```

Why? AFAICT this doesn't allocate or panic.

Reply:
It needs to match `Vec::leak`, which it calls. As for why `Vec::leak` has this, it may be an artifact of the old implementation that *did* allocate.

Reply:
I was just looking at `no_global_oom_handling` for unrelated reasons and came to the same conclusion so I submitted a PR to remove it from `Vec::leak` (#103153).

Reply:
> I was just looking at `no_global_oom_handling` for unrelated reasons and came to the same conclusion so I submitted a PR to remove it from `Vec::leak` (#103153).

Very cool! If your PR is merged before mine, I'll update mine.